crates/bitcoind_rpc/tests/test_filter_iter.rs

```diff
@@ -0,0 +1,142 @@
+    };
+
+    let hash = env.rpc_client().get_best_block_hash()?;
+    let header = env.rpc_client().get_block_header_info(&hash)?;
+    assert_eq!(header.height, 1);
+    let block_1 = BlockId {
+        height: header.height as u32,
+        hash,
+    };
+
+    // `FilterIter` will try to return up to ten recent blocks
+    // so we keep them for reference
+    let new_blocks: Vec<BlockId> = (2..12)
+        .zip(env.mine_blocks(10, None)?)
+        .map(BlockId::from)
+        .collect();
+
+    let new_tip = *new_blocks.last().unwrap();
+
+    struct TestCase {
+        // name
+        name: &'static str,
+        // local blocks
+        chain: Vec<BlockId>,
+        // expected blocks
+        exp: Vec<BlockId>,
+    }
+
+    // For each test we create a new `FilterIter` with the checkpoint given
+    // by the blocks in the test chain. Then we sync to the remote tip and
+    // check the blocks that are returned in the chain update.
+    [
+        TestCase {
+            name: "point of agreement below new blocks, expect base + new",
+            chain: vec![genesis, block_1],
+            exp: [block_1].into_iter().chain(new_blocks.clone()).collect(),
+        },
+        TestCase {
+            name: "point of agreement genesis, expect base + new",
+            chain: vec![genesis],
+            exp: [genesis].into_iter().chain(new_blocks.clone()).collect(),
+        },
+        TestCase {
+            name: "point of agreement within new blocks, expect base + remaining",
+            chain: new_blocks[..=2].to_vec(),
+            exp: new_blocks[2..].to_vec(),
+        },
+        TestCase {
+            name: "stale tip within new blocks, expect base + corrected + remaining",
+            // base height: 4, stale height: 5
+            chain: vec![new_blocks[2], block_id!(5, "E")],
+            exp: new_blocks[2..].to_vec(),
+        },
+        TestCase {
+            name: "stale tip below new blocks, expect base + corrected + new",
+            chain: vec![genesis, block_id!(1, "A")],
+            exp: [genesis, block_1].into_iter().chain(new_blocks).collect(),
+        },
+    ]
+    .into_iter()
+    .for_each(|test| {
+        let cp = CheckPoint::from_block_ids(test.chain).unwrap();
+        let mut iter = FilterIter::new_with_checkpoint(env.rpc_client(), cp);
+        assert_eq!(iter.get_tip().unwrap(), Some(new_tip));
+        let update_cp = iter.chain_update().unwrap();
+        let mut update_blocks: Vec<_> = update_cp.iter().map(|cp| cp.block_id()).collect();
+        update_blocks.reverse();
+        assert_eq!(update_blocks, test.exp, "{}", test.name);
+    });
+
+    Ok(())
+}
+
+#[test]
+fn filter_iter_error_no_scripts() -> anyhow::Result<()> {
+    use bdk_bitcoind_rpc::bip158::Error;
+    let env = testenv()?;
+
+    let addr = env
+        .rpc_client()
+        .get_new_address(None, None)?
+        .assume_checked();
+    let spk = addr.script_pubkey();
+    let _ = env.mine_blocks(9, Some(addr))?;
```

Comment:
If the goal of this test is to check the behavior when no spk has been added to the filter, I would avoid reiterative checks over the same thing. In this case, it would mean to just mine 1 or 2 blocks.

```suggestion
    let _ = env.mine_blocks(1, Some(addr))?;

    let mut iter = FilterIter::new_with_height(env.rpc_client(), 1);
    assert_eq!(iter.get_tip().unwrap().map(|b| b.height), Some(2));

    // iterator should return two errors
    for res in iter.by_ref().take(2) {
        assert!(matches!(res, Err(Error::NoScripts)));
```